src/pages/posts/[postid].astro

```diff
@@ -41,10 +42,22 @@ const html = sanitizeHtml(await marked.parse(content.body), {
 const showCopyLinkWarning = Astro.url.searchParams.get('created') === content.edit_key;
 const url = Astro.url.href.split('?')[0];
 const editURL = url + `/edit?key=${content.edit_key}`;
----
 
+const canViewAdultContent = Astro.url.searchParams.get('adult');
+const showAdultWarning = !canViewAdultContent && content.adult_content;
+---
 <BaseLayout title={`${content.title || 'Untitled'}`}>
     <Seo title={content.title} description={trimmedText} />
+
+    <Modal client:only="preact" isOpen={true} hidePage={true}>
```

Comment:
it's a little unfortunate to have to load preact for every modal, but i'm not sure what a good way around it would be. maybe we can render the modal on the server side, and then only load preact for the buttons? we should discuss